src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseFileService.cs

```diff
@@ -342,6 +342,45 @@ await fileIds.Select(fileId => CheckFileExists(releaseId, fileId, DeletableFileT
                 });
         }
 
+        public Task<Either<ActionResult, FileInfo>> ReplaceAncillary(
+            Guid releaseId,
+            Guid fileId,
+            IFormFile newFile)
+        {
+            return _persistenceHelper
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanUpdateRelease)
+                .OnSuccess(async () => await _fileUploadsValidatorService.ValidateFileForUpload(newFile, Ancillary))
+                .OnSuccess(_ => _persistenceHelper.CheckEntityExists<ReleaseFile>(query =>
+                    query
+                        .Include(rf => rf.File)
+                        .Where(rf => rf.FileId == fileId && rf.ReleaseId == releaseId))
+                )
+                .OnSuccess(async oldReleaseFile =>
+                {
+                    var newReleaseFile = await _releaseFileRepository.Create(
+                        releaseId: releaseId,
+                        filename: newFile.FileName,
+                        contentLength: newFile.Length,
+                        contentType: newFile.ContentType,
+                        type: Ancillary,
+                        createdById: _userService.GetUserId(),
+                        name: oldReleaseFile.Name,
+                        summary: oldReleaseFile.Summary);
+
+                    _contentDbContext.Remove(oldReleaseFile);
```

Comment:
Seems to me like we should remove the old file as well as long as it is no longer linked to any other releases (i.e. amendments). We already do a similar thing in the `Delete` method.

This will also needs some extra tests to verify this behaviour.